packages/validator/src/util/externalSignerClient.ts

```diff
@@ -160,17 +162,68 @@ export async function externalSignerUpCheck(remoteUrl: string): Promise<boolean>
     headers: {"Content-Type": "application/json"},
   });
 
-  const data = await handlerExternalSignerResponse<{status: string}>(res);
+  const data = await handlerExternalSignerUpCheckResponse(res);
   return data.status === "OK";
 }
 
-async function handlerExternalSignerResponse<T>(res: Response): Promise<T> {
+async function handlerExternalSignerGetKeysResponse<T>(res: Response, logger?: Logger): Promise<string[]> {
   if (!res.ok) {
     const errBody = await res.text();
+    logger?.debug("External signer GetKeys error", {body: errBody});
```

Comment:
Isn't this error already logged upstream?

Reply:
the error message looks you posted looks strange, why is there a nested errorBody?

Reply:
I'm not sure why, but the exception body was displayed the same way before my change:

`Mar-28 15:42:28.014[]                error: Error signing attestation slot=1311512, index=27, head=0xb9dcc954f6393a71d28cb853f81481eef62b14c5dce6fb492d8df965b517055a, validatorIndex=1626752 -
Error:
    at externalSignerPostSignature (file:///usr/app/packages/validator/src/util/externalSignerClient.ts:159:11)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)...`

I just added some context and the status code. In these particular cases the response returns an empty body. If there was some content then it is displayed in both places right after the velidatorIndex and the Error: label.

Should I leave the context and response status code in the message or should I change it back to what it was before?

Reply:
it's pretty bad that the server doesn't not return an error, is this specific to diva remote signer? I would have to see how web3signer responds if there is an error.

Since the usage is very limited of these functions and we know where those are used might be sufficient to throw a simple error, upstream log already adds context. but again would have to test how it works with web3signer which is what 99% of users use at the moment, need to make sure we keep the UX clean but at the same time provide sufficient information to debug failures

Reply:
Maybe check if body is empty and if it is, use response status code as exception message?

Reply:
Looks like `res.statusText` is also empty:
`Mar-28 16:49:14.085[]                error: Error signing attestation slot=1311846, index=7, head=0x344b807c111cdefe34f573f33590f87f6fd56d09759dde76e75e6e1e4550acd3, validatorIndex=1626972 -
Error:
    at externalSignerPostSignature (file:///usr/app/packages/validator/src/util/externalSignerClient.ts:156:11)...`

Reply:
I think it's due to https://github.com/ChainSafe/lodestar/pull/6587#discussion_r1543341186